### PR TITLE
Add cxx interop build pipeline

### PR DESCRIPTION
**Pull request by gjcairo**
This PR adds a new target to `docker-compose` that builds this project with C++ interoperability enabled, with the goal of adding a new CI pipeline that performs this check.

### Motivation:

To make sure that C++ interoperability can be used on projects that depend on `swift-crypto`.

### Modifications:

- Added a new target to docker-compose that builds this project with C++ interoperability enabled.
- Will request a new CI pipeline that runs this new target.

### Result:

We will ensure this projects builds properly with C++ interoperability enabled.

### CODE REVIEW

**Comment by gjcairo** [comment]
@yim-lee would you mind setting up a new `pull request validation (cxx interop)` pipeline that runs the new target?

File: docker/docker-compose.yaml [hunk 1]
```diff
@@ -35,6 +35,10 @@ services:
   cmake:
     <<: *common
     command: /bin/bash -xcl "cmake -G Ninja -D CMAKE_BUILD_TYPE=Release -B out -S . && ninja -C out"
+    
+  cxx-interop-build:
```
Comment (yim-lee) [inline docker/docker-compose.yaml:39]: Need to add `cxx-interop-build` to `docker-compose.2204.59.yaml` etc. to override image.

**Comment by yim-lee** [comment]
@swift-server-bot add to allowlist

**Comment by yim-lee** [comment]
@gjcairo I added new pipeline `pull request validation (cxx interop) pipeline` as requested. It uses Swift 5.9, similar to "api breakage" and "cmake" pipelines. We can set up separate "cxx interop" pipelines for Swift 5.9 and 5.10 if you want. Just let me know.

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by gjcairo** [comment]
Thanks @yim-lee ! I had to make some changes to the step so hopefully it all still works :D 
I think it would make sense to have a pipeline for 5.10 in this case, as I suppose things related to the interoperability may change between Swift versions and we want to make sure we are still green with newer versions. Would you mind setting up a second pipeline? I have updated the corresponding docker-compose file for 5.10 too.

**Comment by yim-lee** [comment]
CI updated.

@swift-server-bot test this please

**Review by glbrntt — APPROVED** [review]

**Review by Lukasa — APPROVED** [review]

**Pull request merged**